BinaryObjectScanner/Protection/CopyX.cs

```diff
@@ -0,0 +1,195 @@
+
+        /// <inheritdoc/>
+        public string? CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
+        {
+
+            // Checks for Professional
+            // PEX checks intentionally only detect Professional
+
+            var sections = pex.Model.SectionTable;
+            if (sections == null)
+                return null;
+
+            if (pex.OverlayStrings != null)
+            {
+                // Checks if main executable contains reference to optgraph.dll. 
+                // This might be better removed later, as Redump ID 82475 is a false positive, and also doesn't actually
+                // contain the actual optgraph.dll file.
+                // TODO: Find a way to check for situations like Redump ID 48393, where the string is spaced out with
+                // 0x00 between letters and does not show up on string checks.
+                // TODO: This might need to check every single section. Unsure until more samples are acquired.
+                // TODO: TKKG also has an NE 3.1x executable with a reference. This can be added later.
+                // Samples: Redump ID 108150
+                if (pex.OverlayStrings.Any(s => s.Contains("optgraph.dll")))
+                    return "copy-X";
+            }
+
+            var strs = pex.GetFirstSectionStrings(".rdata");
+            if (strs != null)
+            {
+                // Samples: Redump ID 82475, German Emergency 2 Deluxe, Redump ID 48393
+                if (strs.Any(s => s.Contains("optgraph.dll")))
+                    return "copy-X";
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc/>
+#if NET20 || NET35
+        public Queue<string> CheckDirectoryPath(string path, IEnumerable<string>? files)
+#else
+        public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string>? files)
+#endif
+        {
+#if NET20 || NET35
+            var protections = new Queue<string>();
+#else
+            var protections = new ConcurrentQueue<string>();
+#endif
+
+            // Checks for Light
+            // Directory checks intentionally only detect Light
+
+            if (files == null)
+                return protections;
+
+            // Gets files in ZDAT*
+            // Excludes files with .x64 extension to avoid flagging Professional files.
+            var fileList = files.Where(f => !f.EndsWith(".x64", StringComparison.OrdinalIgnoreCase))
+                .Where(f =>
+                {
+                    // TODO: Compensate for the check being run a directory or more higher
+                    f = f.Remove(0, path.Length);
+                    f = f.TrimStart('/', '\\');
+                    return f.StartsWith("ZDAT", StringComparison.OrdinalIgnoreCase);
+                })
+
+                // Sorts list of files in ZDAT* so just the first file gets pulled, later ones have a chance of the ring
```

Comment:
Please don't include comments halfway through a method chain like this. Either move it to the top of the chain or add it after.